lib/facter/ec2.rb

```diff
@@ -46,8 +46,15 @@
     end
 
     setcode do
-      @querier.fetch
+      userdata = @querier.fetch
+      begin
+        userdata.encode(Encoding::UTF_8)
+      rescue EncodingError
```

Comment:
Apparently encoding handling is a problem in Ruby 1.8.7; I think it'd be ok to only do this check for Ruby 1.9+ /cc @joshcooper